lib/active_shipping/carriers/ups.rb

```diff
@@ -998,6 +1000,10 @@ def response_message(document)
       [status, desc].select(&:present?).join(": ").presence || "UPS could not process the request."
     end
 
+    def rate_warning_messages(rate_xml)
+      rate_xml.xpath("RatedShipmentWarning").map { |warning| warning.text }
```

Comment:
Will this return an empty set or `nil` if no elements match? Do we have an existing test that covers this?